Attributes/GithubRepositoryAttribute.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.Text.RegularExpressions;
+
+namespace GeekHub.Attributes
+{
+    public class GithubRepositoryAttribute : ValidationAttribute
+    {
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            if (value is null || string.IsNullOrWhiteSpace(value.ToString())) return ValidationResult.Success;
```

Comment:
It is an optional attribute.

Reply:
Although I should probably remove `string.IsNullOrWhiteSpace(...)` hmm

Reply:
I don't think it makes a difference whether it's optional or not? Unless a null or whitespace value is considered valid in our context, this seems like it will be quite confusing.

Reply:
It's to be consistent with the other DataAnnotation attributes, if the parameter is required, it should be indicated with `[Required]`.